src/Symfony/Component/Form/Extension/Validator/Constraints/FormValidator.php

```diff
@@ -48,7 +48,7 @@ public function validate($form, Constraint $constraint)
 
             // Validate the data against its own constraints
             if ($form->isRoot() && (\is_object($data) || \is_array($data))) {
-                if (\is_array($groups) && \count($groups) > 0 || $groups instanceof GroupSequence && \count($groups->groups) > 0) {
+                if (($groups && \is_array($groups)) || ($groups instanceof GroupSequence && $groups->groups)) {
```

Comment:
unnecessary braces have been added

Reply:
Added with a purpose of clarifying the logical intention =)